idaes/core/control_volume0d.py

```diff
@@ -1531,112 +1531,243 @@ def _get_stream_table_contents(self, time_point=0):
+                if iscale.get_scaling_factor(v) is None:
+                    sf = iscale.get_scaling_factor(self.volume[t])
+                    sf *= iscale.get_scaling_factor(self.phase_fraction[t, p])
+                    sf *= iscale.get_scaling_factor(
+                        self.properties_out[t].get_material_density_terms(p, j),
+                        default=1, warning=True)
+                    iscale.set_scaling_factor(v, sf)
 
         if hasattr(self, "material_accumulation"):
-            for i, v in self.material_accumulation.items():
-                sf = 100*iscale.get_scaling_factor(
-                    self.material_holdup[i], default=1, warning=True)
+            for (t, p, j), v in self.material_accumulation.items():
+                if iscale.get_scaling_factor(v) is None:
+                    sf = 100 * iscale.get_scaling_factor(
+                        self.material_holdup[t, p, j])
+                    iscale.set_scaling_factor(v, sf)
+
+        if hasattr(self, "rate_reaction_generation"):
+            for (t, p, j), v in self.rate_reaction_generation.items():
+                if iscale.get_scaling_factor(v) is None:
+                    sf = iscale.get_scaling_factor(
+                        self.properties_out[t].get_material_flow_terms(p, j),
+                        default=1, warning=True)
+                    iscale.set_scaling_factor(v, sf)
+
+        if hasattr(self, "equilibrium_reaction_generation"):
+            for (t, p, j), v in self.equilibrium_reaction_generation.items():
+                if iscale.get_scaling_factor(v) is None:
+                    sf = iscale.get_scaling_factor(
+                        self.properties_out[t].get_material_flow_terms(p, j),
+                        default=1, warning=True)
+                    iscale.set_scaling_factor(v, sf)
+
+        if hasattr(self, "phase_equilibrium_generation"):
+            for (t, e), v in self.phase_equilibrium_generation.items():
+                if iscale.get_scaling_factor(v) is None:
+                    sf = iscale.min_scaling_factor(
+                        [self.properties_out[t].get_material_flow_terms(p, j)
+                         for p in phase_list])
+                    iscale.set_scaling_factor(v, sf)
+
+        if hasattr(self, "mass_transfer_term"):
+            for (t, p, j), v in self.mass_transfer_term.items():
+                if iscale.get_scaling_factor(v) is None:
+                    sf = iscale.get_scaling_factor(
+                        self.properties_out[t].get_material_flow_terms(p, j),
+                        default=1, warning=True)
+                    iscale.set_scaling_factor(v, sf)
+
+        if hasattr(self, "rate_reaction_extent"):
+            if iscale.get_scaling_factor(self.rate_reaction_extent) is None:
+                # extent of reaction typically between 0.01 and 1
+                iscale.set_scaling_factor(self.rate_reaction_extent, 10)
```

Comment:
Reaction extent is equal to the reaction rate multiplied by the stoichiometric coefficient. Given the coefficients are generally only order 1, this should be equal to the scaling factor of the reaction rate from the Reaction Block.

Reply:
Oh I didn't know what reaction extent was, I assumed it was the standard term "extent of reaction" which goes from 0 to 1.